code/DynamicCache.php

```diff
@@ -92,14 +92,33 @@ protected function enabled($url)
         $basicAuthConfig = Config::inst()->forClass('BasicAuth');
         if($basicAuthConfig->entire_site_protected) {
             $member = null;
+
+            // NOTE(Jake): Required so BasicAuth::requireLogin() doesn't early exit with a 'true' value
+            //             This will affect caching performance with BasicAuth turned on.
+            if(!Security::database_is_ready()) {
```

Comment:
You should use !DB::is_active(), since database_is_ready() does lots of unecessary schema checks as well.